admin/sql/eaa/CreateFunctions.sql

```diff
@@ -0,0 +1,72 @@
+BEGIN;
+
+SET search_path = 'event_art_archive';
+
+CREATE OR REPLACE FUNCTION materialize_eaa_presence() RETURNS trigger AS $$
+    BEGIN
+        -- On delete, set the presence flag to 'absent' if there's no more
+        -- event art
+        IF TG_OP = 'DELETE' THEN
+            IF NOT EXISTS (
+                SELECT TRUE FROM event_art_archive.event_art
+                WHERE event = OLD.event
+            ) THEN
+                UPDATE musicbrainz.event_meta
+                SET event_art_presence = 'absent'
+                WHERE id = OLD.event;
+            END IF;
+        END IF;
+
+        -- On insert, set the presence flag to 'present' if it was previously
+        -- 'absent'
+        IF TG_OP = 'INSERT' THEN
+            CASE (
+                SELECT event_art_presence FROM musicbrainz.event_meta
+                WHERE id = NEW.event
+            )
+                WHEN 'absent' THEN
+                    UPDATE musicbrainz.event_meta
+                    SET event_art_presence = 'present'
+                    WHERE id = NEW.event;
+                WHEN 'darkened' THEN
+                    RAISE EXCEPTION 'This event has been darkened and cannot have new event art';
```

Comment:
This is less obviously true for events than releases (if a poster has been taken down, that doesn't necessarily mean a ticket wouldn't be ok to upload) but I guess darkening is rare enough it shouldn't matter anyway.

Reply:
That's true...though from what I understand, if it's darkened on the IA side, uploading any artwork to there wouldn't work, so the exception would also be technically true.